src/TLuaInterpreter.cpp

```diff
@@ -10449,13 +10449,547 @@ int TLuaInterpreter::openWebPage(lua_State* L)
     if (lua_isstring(L, 1)) {
         QString url = lua_tostring(L, 1);
         lua_pushboolean(L, mudlet::self()->openWebPage(url));
+        return 1;
     } else {
         lua_pushfstring(L, "openWebPage: bad argument #%d (string expected, got %s)", 1, luaL_typename(L, 1));
-        lua_error(L);
+        return lua_error(L);
+    }
+}
+
+/*
+ * Overrides the Discord application id used for this profile
+ * By definition this stops the ability to use the Icon resources that have been
+ * set up for the Mudlet client - so that ones specific to, say, a MUD Server
+ * can be used INSTEAD.
+ *
+ * The Application ID seems to be an 18 digit number at the moment.
+ * Specifying an empty string or no argument resets to using the default Mudlet
+ * one...
+ *
+ * There is no lua getter for this value, if it is supplied via a GMCP or other
+ * OOB from a Server they may not want to advertise it - though currently it
+ * should be readable from the "Special options" tab of the
+ * "Profile preferences".
+ */
+// TODO: Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setDiscordApplicationID
+int TLuaInterpreter::setDiscordApplicationID(lua_State* L)
+{
+    mudlet* pMudlet = mudlet::self();
+    auto& host = getHostFromLua(L);
+
+    auto result = discordApiEnabled(L);
```

Comment:
This should be a write access, I believe.

Reply:
Fixed